### PR TITLE
Continuous improvement extensions

### PR DESCRIPTION
**Pull request by ivorc**

### CODE REVIEW

File: continuous-improvement.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ It takes continuous effort to maintain and evolve processes in response to chall
 
 ### Control technical debt
 
-Technical debt arises due to processes or practices in the past, but has an ongoing impact on the present.
+Technical debt is a term which refers to things about the way a system is built which are not apparent to users of the system, but impact the ability of the team to make changes to it quickly and safely. Tech debt arises due to processes or practices in the past, but has an ongoing impact on the present.
 
 Technical debt:
 *	Leads to bugs and loss of reliability.
```
Comment (andyblundell) [inline continuous-improvement.md:60]: Maybe call out the pain of working on the system, too?

... ah just read the next paragraph :)

File: continuous-improvement.md [hunk 2]
```diff
@@ -64,11 +64,26 @@ Technical debt:
 
 Without sustained improvement effort these all get worse over time, reducing capacity to deliver features. If little or no tech debt improvement work is done, delivery may initially be faster but over time it becomes progressively slower and less predictable.
 
+### Improve reliability and operability
+
+Some important improvement work consistens of implementing or improving reliability or operability features, such as monitoring dashboards, application logs or automated security testing.
```
Comment (andyblundell) [inline continuous-improvement.md:69]: "consistens"

File: continuous-improvement.md [hunk 3]
```diff
@@ -64,11 +64,26 @@ Technical debt:
...
+
+Where things like this are missing or need improvement, it is not tech debt &mdash; improving them brings a tangible benefit to service reliability, which users do notice and care about. As such, this work should be treated like any other feature work and prioritised alongside functional features.
```
Comment (andyblundell) [inline continuous-improvement.md:71]: Suggests tech debt *shouldn't* make it on the backlog?

File: continuous-improvement.md [hunk 4]
```diff
@@ -160,9 +190,40 @@ In addition, it is useful to track the proportion of time being spent on various
   * Tech debt
   * Other improvement work
 
+## What does this mean for me as a less-technical person?
```
Comment (andyblundell) [inline continuous-improvement.md:193]: As per chat, let's take this bit out for now and discuss the wording with the product profession leads

File: continuous-improvement.md [hunk 5]
```diff
@@ -160,9 +190,40 @@ In addition, it is useful to track the proportion of time being spent on various
   * Tech debt
```
Comment (andyblundell) [inline continuous-improvement.md:190]: Did you mean to drop the "proportion of time being spent on"?

File: continuous-improvement.md [hunk 6]
```diff
@@ -88,9 +103,18 @@ Examples include:
 * Technologies which should be replaced.
 * Areas with insufficient, inefficient or ineffective testing.
 
+### 3. Reliability and operability
```
Comment (andyblundell) [inline continuous-improvement.md:106]: 3? :)

**Review by andyblundell — COMMENTED** [review]
Should we mention hypothesis-based experiments, and include the template we're suggesting for how to describe an experiment?

**Comment by arctangent** [comment]
You might be interested in this short interview with Ward Cunningham (who invented the technical debt metaphor): https://www.youtube.com/watch?v=pqeJFYwnkjE

He describes technical debt as being the difference between (1) what you would have done had you known what you were doing all along (i.e. with 20-20 hindsight) and (2) what you currently have (i.e. a system that has evolved to meet changing requirements/constraints during the course of development).

Perhaps it is worth mentioning somewhere that the difficulty in updating/maintaining an un-refactored system is because - at least in part - the system does not encode an accurate understanding of the problem domain or the "hindsight solution"?

Perhaps it is also worth distinguishing between (A) tech debt which is intentionally accrued (e.g. "build one to throw away" during an architectural spike, or relaxing test coverage standards) in order to learn more faster or to meet a hard deadline and (B) unintentional/emergent tech debt which becomes apparent when requirements/constraints change.

Finally, as Ward suggests in the video, poor quality code should never be accepted under the intentionally accrued tech debt category, and in fact poor quality code makes paying down tech debt more difficult. Perhaps it is worth mentioning that continuous improvement is easiest to do when the quality of what you do have is already high (a.k.a. the "you can't polish a turd" pattern).

File: continuous-improvement.md (hunk 5 above)
Comment (ivorc) [reply continuous-improvement.md:190]: It's not been dropped. Just a diff oddity, I think.

**Review by ivorc — COMMENTED** [review]
Review comment responses.

**Comment by ivorc** [comment]
@arctangent — all good stuff around the details of tech debt. I'll get this merged as is for now, but open to a PR to add to it, so long as it doesn't inflate it too much.

**Comment by ivorc** [comment]
@andyblundell we do have a template under 
> Express each experiment as a hypothesis:
in the Acting section

**Review by andyblundell — APPROVED** [review]

**Pull request merged**